### ISSUE TITLE
`bundle install` ignoring Gemfile

### ISSUE THREAD

**Issue by Joshfindit** [bundler/bundler#5233]
### Resolved this while gathering info: Submitting just in case someone else gets this issue.

**TL;DR**: http://bradgessler.com/articles/docker-bundler/ has a list of steps, including hard-setting the Gemfile via "ENV BUNDLE_GEMFILE=$APP_HOME/Gemfile". This got set to `/Gemfile` which ignored `./Gemfile`

----

I created a docker container to hold a rails app, and it builds/runs fine.
When changing the gems in the Gemfile and running `bundle update`, it completely ignores the Gemfile in the same directory.

### System Info

`ruby --version
ruby 2.3.3p222 (2016-11-21 revision 56859) [x86_64-linux]`
`bundle --version
Bundler version 1.13.6`
`uname -a
Linux 59601804b428 4.4.27-moby #1 SMP Wed Oct 26 14:21:29 UTC 2016 x86_64 GNU/Linux`
`ENV BUNDLE_GEMFILE=/Gemfile`

### Dockerfile

```
FROM ruby:latest

RUN apt-get update -qq && apt-get install -y build-essential

# for nokogiri
RUN apt-get install -y libxml2-dev libxslt1-dev

# for a JS runtime
RUN apt-get install -y nodejs

#NV BUNDLE_GEMFILE=$APP_HOME/Gemfile \
#  BUNDLE_JOBS=2 \
#  BUNDLE_PATH=/bundle
ENV BUNDLE_GEMFILE=$APP_HOME/Gemfile #Side note: this doesn't seem to work at all. It's based on a blog post.

RUN gem install bundler

ADD RailsApp/Gemfile* $APP_HOME/ 


ENV APP_HOME /root/RailsApp
RUN if [ ! -d "$APP_HOME" ]; then mkdir $APP_HOME; fi
WORKDIR $APP_HOME

RUN bundle install
```

### docker-compose
```
version: '2'
services:
  railstest:
    build: .
    command: bin/rails server --port 3000 --binding 0.0.0.0
    # command: bundle install
    ports:
      - "3000:3000"

    volumes:
      - ./RailsApp:/root/RailsApp

volumes:
  railstest:
```

### Gemfile

```
source 'https://rubygems.org'

git_source(:github) do |repo_name|
  repo_name = "#{repo_name}/#{repo_name}" unless repo_name.include?("/")
  "https://github.com/#{repo_name}.git"
end


# Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
gem 'rails', '>= 5.0.1.rc1', '< 5.1'
# Use sqlite3 as the database for Active Record
# gem 'sqlite3'
# Use Puma as the app server
gem 'puma', '~> 3.0'
# Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
# gem 'jbuilder', '~> 2.5'
# Use Redis adapter to run Action Cable in production
# gem 'redis', '~> 3.0'
# Use ActiveModel has_secure_password
# gem 'bcrypt', '~> 3.1.7'

# Use Capistrano for deployment
# gem 'capistrano-rails', group: :development

# Use Rack CORS for handling Cross-Origin Resource Sharing (CORS), making cross-origin AJAX possible
# gem 'rack-cors'
gem 'neo4j'

group :development, :test do
  # Call 'byebug' anywhere in the code to stop execution and get a debugger console
  gem 'byebug', platform: :mri
end

group :development do
  gem 'listen', '~> 3.0.5'
  # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
  gem 'spring'
  gem 'spring-watcher-listen', '~> 2.0.0'
end

# Windows does not include zoneinfo files, so bundle the tzinfo-data gem
gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]


gem 'redcarpet' # Markdown render
gem 'pygments.rb' # Syntax highlighting for redcarpet. Needs `$ pygmentize -S default -f html > style.css` to be run
gem "neo4jrb-paperclip", :require => "neo4jrb_paperclip"

gem 'plist' #For reading DayOne Entries
gem 'os' #For querying about which OS we're running on
```

### Output of `bundle install`

```
Using rake 12.0.0
Using concurrent-ruby 1.0.2
Using i18n 0.7.0
Using minitest 5.10.1
Using thread_safe 0.3.5
Using builder 3.2.2
Using erubis 2.7.0
Using mini_portile2 2.1.0
Using rack 2.0.1
Using nio4r 1.2.1
Using websocket-extensions 0.1.2
Using mime-types-data 3.2016.0521
Using arel 7.1.4
Using byebug 9.0.6
Using coffee-script-source 1.11.1
Using execjs 2.7.0
Using method_source 0.8.2
Using thor 0.19.4
Using debug_inspector 0.0.2
Using ffi 1.9.14
Using multi_json 1.12.1
Using rb-fsevent 0.9.8
Using puma 3.6.2
Using bundler 1.13.6
Using sass 3.4.22
Using tilt 2.0.5
Using sqlite3 1.3.12
Using turbolinks-source 5.0.0
Using tzinfo 1.2.2
Using nokogiri 1.6.8.1
Using rack-test 0.6.3
Using sprockets 3.7.0
Using websocket-driver 0.6.4
Using mime-types 3.1
Using coffee-script 2.4.1
Using uglifier 3.0.4
Using rb-inotify 0.9.7
Using turbolinks 5.0.1
Using activesupport 5.0.1.rc1
Using loofah 2.0.3
Using mail 2.6.4
Using listen 3.0.8
Using rails-dom-testing 2.0.1
Using globalid 0.3.7
Using activemodel 5.0.1.rc1
Using jbuilder 2.6.1
Using spring 2.0.0
Using rails-html-sanitizer 1.0.3
Using activejob 5.0.1.rc1
Using activerecord 5.0.1.rc1
Using spring-watcher-listen 2.0.1
Using actionview 5.0.1.rc1
Using actionpack 5.0.1.rc1
Using actioncable 5.0.1.rc1
Using actionmailer 5.0.1.rc1
Using railties 5.0.1.rc1
Using sprockets-rails 3.2.0
Using coffee-rails 4.2.1
Using jquery-rails 4.2.1
Using web-console 3.4.0
Using rails 5.0.1.rc1
Using sass-rails 5.0.6
Bundle complete! 15 Gemfile dependencies, 62 gems now installed.
Bundled gems are installed into /usr/local/bundle.
```

### Output of `bundle install --gemfile=Gemfile`

```
Using rake 12.0.0
Using concurrent-ruby 1.0.2
Using i18n 0.7.0
Using minitest 5.10.1
Using thread_safe 0.3.5
Using builder 3.2.2
Using erubis 2.7.0
Using mini_portile2 2.1.0
Using rack 2.0.1
Using nio4r 1.2.1
Using websocket-extensions 0.1.2
Using mime-types-data 3.2016.0521
Using arel 7.1.4
Using byebug 9.0.6
Using colored 1.2
Using multipart-post 2.0.0
Using multi_json 1.12.1
Using ffi 1.9.14
Using multi_xml 0.6.0
Using httpclient 2.8.2.4
Using json 2.0.2
Using rb-fsevent 0.9.8
Using method_source 0.8.2
Using mimemagic 0.3.0
Using neo4j-community 2.2.4.1
Using os 0.9.6
Using ruby-progressbar 1.8.1
Using rubyzip 1.2.0
Using net-http-persistent 2.9.4
Using orm_adapter 0.5.0
Using plist 3.2.0
Using posix-spawn 0.3.12
Using puma 3.6.2
Using yajl-ruby 1.2.1
Using bundler 1.13.6
Using thor 0.19.4
Using redcarpet 3.3.4
Using sqlite3 1.3.12
Using tzinfo 1.2.2
Using nokogiri 1.6.8.1
Using rack-test 0.6.3
Using sprockets 3.7.0
Using websocket-driver 0.6.4
Using mime-types 3.1
Using faraday 0.9.2
Using rb-inotify 0.9.7
Using httparty 0.14.0
Using neo4j-rake_tasks 0.7.9
Using pygments.rb 0.6.3
Using activesupport 5.0.1.rc1
Using loofah 2.0.3
Using mail 2.6.4
Using faraday_middleware 0.10.1
Using listen 3.0.8
Using rails-dom-testing 2.0.1
Using globalid 0.3.7
Using activemodel 5.0.1.rc1
Using climate_control 0.0.3
Using spring 2.0.0
Using rails-html-sanitizer 1.0.3
Using faraday_middleware-multi_json 0.0.6
Using activejob 5.0.1.rc1
Using activerecord 5.0.1.rc1
Using cocaine 0.5.8
Using spring-watcher-listen 2.0.1
Using actionview 5.0.1.rc1
Using neo4j-core 6.1.6
Using paperclip 4.3.7
Using actionpack 5.0.1.rc1
Using neo4j 7.2.3
Using actioncable 5.0.1.rc1
Using actionmailer 5.0.1.rc1
Using railties 5.0.1.rc1
Using sprockets-rails 3.2.0
Using neo4jrb-paperclip 0.0.3
Using rails 5.0.1.rc1
Bundle complete! 14 Gemfile dependencies, 76 gems now installed.
Bundled gems are installed into /usr/local/bundle.
```

**Comment by Joshfindit** [comment]
resolved: `ENV` variable `BUNDLE_GEMFILE` set to the wrong file.
